packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js

```diff
@@ -7772,6 +7773,116 @@ QUnit.module('label integration', () => {
+        const $parentTagContainerIds = $tagBox.attr('aria-labelledby').split(' ');
+
+        assert.strictEqual($parentTagContainerIds.length, 2, 'root element aria-labelledby attribute contains two ids');
+        assert.strictEqual($parentTagContainerIds.includes($multiTagContainer.attr('id')), true, 'aria-labelledby attribute contains multitag id');
+    });
+
+    QUnit.test('check if tagBox is showing the right count of tags correctly if showMultiTagOnly=true and maxDisplay tags is 2', function(assert) {
+        const sampleData = ['First Item', 'Second Item', 'Third Item', 'Fourth Item'];
+        const $tagBox = $('#tagBox').dxTagBox({
+            items: sampleData,
+            value: sampleData,
+            maxDisplayedTags: 2,
+            showMultiTagOnly: true,
+        });
+        const $parentTagContainerIds = $tagBox.attr('aria-labelledby').split(' ');
+        const $tagIds = $tagBox.find('.dx-tag').map((_, element) => $(element).attr('id')).get();
+
+        assert.strictEqual($parentTagContainerIds.every(item => $tagIds.includes(item)), true, 'root element contains all tag ids when showMultiTagOnly = true and maxDisplayTags = 2');
+    });
+
+    QUnit.test('check if tagBox is showing the right count of tags correctly if showMultiTagOnly=true and maxDisplayTags is 4', function(assert) {
+        const sampleData = ['First Item', 'Second Item', 'Third Item', 'Fourth Item'];
+        const $tagBox = $('#tagBox').dxTagBox({
+            items: sampleData,
+            value: sampleData,
+            maxDisplayedTags: 4,
+            showMultiTagOnly: true,
+        });
+        const $parentTagContainerIds = $tagBox.attr('aria-labelledby').split(' ');
+        const $tagIds = $tagBox.find('.dx-tag').map((_, element) => $(element).attr('id')).get();
+
+        assert.strictEqual($parentTagContainerIds.every(item => $tagIds.includes(item)), true, 'root element contains all tag ids when showMultiTagOnly = true and maxDisplayTags = 4');
+    });
+
+    QUnit.test('check if tagBox is showing the right count of tags correctly if showMultiTagOnly=false and maxDisplayTags is 2', function(assert) {
+        const sampleData = ['First Item', 'Second Item', 'Third Item', 'Fourth Item'];
+        const $tagBox = $('#tagBox').dxTagBox({
+            items: sampleData,
+            value: sampleData,
+            maxDisplayedTags: 2,
+            showMultiTagOnly: false,
+        });
+        const $parentTagContainerIds = $tagBox.attr('aria-labelledby').split(' ');
+        const $tagIds = $tagBox.find('.dx-tag').map((_, element) => $(element).attr('id')).get();
+
+        assert.strictEqual($parentTagContainerIds.every(item => $tagIds.includes(item)), true, 'root element contains all tag ids when showMultiTagOnly = false and maxDisplayTags = 2');
+    });
+
+    QUnit.test('check if tagBox is showing the right count of tags correctly if showMultiTagOnly=false and maxDisplayTags is 4', function(assert) {
```

Comment:
Hm, what's the difference between this scenario and scenario in the previous tests?
Seems like this test cannot be red if the previous one is green.

Reply:
The difference between two test is that this test as you can see the max display is 4 and the item example has 4 items in total.
The previous test its 2 so the tag count should be 2 exactly because maxdisplay is 2 
so in this test its max display = 4 so the expected max display should be 4 simple tags because it must display all items in the items array.

Reply:
I guess i should add an additional assertion to test that the tags in this scenario should not be a multitag but just an ordinary tag